sda/cmd/s3inbox/proxy.go

```diff
@@ -223,6 +228,24 @@ func (p *Proxy) allowedResponse(w http.ResponseWriter, r *http.Request) {
 	_ = s3response.Body.Close()
 }
 
+// Renew the connection to MQ if necessary, then send message
+func (p *Proxy) checkAndSendMessage(jsonMessage []byte, r *http.Request) error {
+	if p.messenger.IsConnClosed() {
+		log.Warning("connection is closed, reconnecting...")
+		m, err := broker.NewMQ(Conf.Broker)
+		p.messenger = m
+
+		return err
+	}
+
+	if err := p.messenger.SendMessage(p.fileIds[r.URL.Path], p.messenger.Conf.Exchange, p.messenger.Conf.RoutingKey, jsonMessage); err != nil {
+
+		return fmt.Errorf("error when sending message to broker: %v", err)
+	}
+
+	return nil
+}
+
```

Comment:
There is no need to have this as a separate function since it is only used once.
But there are a case missing, when the channel is closed.

```
		if p.messenger.Connection.IsClosed() {
			log.Warning("connection is closed, reconnecting...")
			p.messenger, err = broker.NewMQ(p.messenger.Conf)
			if err != nil {
				log.Errorf("broker error: %v", err)
				reportError(http.StatusInternalServerError, "could not connect to broker", w)

				return
			}
		}
		if p.messenger.Channel.IsClosed() {
			log.Warning("channel is closed, recreating...")
			if err := p.messenger.CreateNewChannel(); err != nil {
				log.Errorf("broker error: %v", err)
				reportError(http.StatusInternalServerError, "could not create channel", w)

				return
			}
		}

		if err := p.messenger.SendMessage(p.fileIds[r.URL.Path], p.messenger.Conf.Exchange, p.messenger.Conf.RoutingKey, jsonMessage); err != nil {
			log.Errorf("sending message failed: %v", err)
			reportError(http.StatusInternalServerError, "communication failure", w)

			return
		}
```

There are also tests missing for the cases where the channel and connection is closed.